src/components/BookingModal/BookingForm/index.js

```diff
@@ -0,0 +1,168 @@
+import React, { Fragment } from 'react';
+import { PropTypes as PT } from 'prop-types';
+import container from './container';
+import { Form, Input } from '../../common';
+import eventTypes from '../../../utilities/eventTypes';
+import holidayStatus from '../../../utilities/holidayStatus';
+
+const BookingForm = props => {
+  const {
+    isEventBeingUpdated,
+    createEvent,
+    updateEvent,
+    formData,
+    formStatus,
+    formIsValid,
+    bookingDuration,
+    booking,
+    workingFromHomeBooking,
+    submitButtonDisabled,
+    availableDays,
+  } = props;
+
+  const {
+    eventStatus: { eventStatusId },
+  } = booking;
+  const isEventCancelled = eventStatusId === holidayStatus.CANCELLED;
+
+  const { eventTypeId } = formData;
+
+  const createCtas = () => {
+    const buttonTextValue = () => {
+      const durationText = bookingDuration === 0.5 ? 'Half' : bookingDuration;
+      const dayText = bookingDuration > 1 ? 'Days' : 'Day';
+
+      if (isEventBeingUpdated) {
+        if (isEventCancelled) {
+          return 'Cancelled';
+        }
+        return `Update to ${durationText} ${dayText}`;
+      } else {
+        if (eventTypeId !== eventTypes.ANNUAL_LEAVE) {
+          return 'Request WFH';
+        }
+        return `Request ${durationText} ${dayText}`;
+      }
+    };
+
+    let isDisabled = false;
+    if (!isEventBeingUpdated) {
+      isDisabled = bookingDuration > availableDays;
+    }
+
+    if (isEventBeingUpdated) {
+      return [
+        {
+          label: buttonTextValue(),
+          event: updateEvent,
+          disabled: isEventCancelled || submitButtonDisabled,
+        },
+      ];
+    } else {
+      return [
+        {
+          label: buttonTextValue(),
+          event: createEvent,
+          disabled: isDisabled,
+        },
+      ];
+    }
+  };
+
+  const renderWFH = () => {
+    const options = [
+      { value: 1, displayValue: 'Annual Leave' },
+      { value: 2, displayValue: 'Working from home' },
+    ];
+    workingFromHomeBooking ? options.shift() : '';
```

Comment:
i'm not mad about using `options.shift()` here - js is a functional language and you should stick with pure functions as much as possible, but I'll leave this up to you :)